scripting/src/main/java/org/infinispan/scripting/impl/ScriptMetadataParser.java

```diff
@@ -0,0 +1,102 @@
+package org.infinispan.scripting.impl;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.StringReader;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.infinispan.scripting.logging.Log;
+import org.infinispan.util.logging.LogFactory;
+
+public class ScriptMetadataParser {
+   private static final Log log = LogFactory.getLog(ScriptMetadataParser.class, Log.class);
+   private static final String DEFAULT_SCRIPT_EXTENSION = "js";
+   private static final Pattern METADATA_COMMENT = Pattern.compile("^(?:#|//|;;)\\s*(.+)");
+   private static final Pattern METADATA_KV = Pattern
+         .compile("\\s*(\\w+)\\s*=\\s*(\"[^\"]*\"|\'[^\']*\'|\\[[\\w,\\s]*\\]|[^,=\\s\"]+)\\s*,?");
```

Comment:
TBH it's a bit harder to read after the update :)

I thought I already wrote this, but could you add a comment with some examples of the regexp is supposed to match? It would make it a little easier to read. You should probably also add some user-facing documentation about the metadata syntax, if you do maybe you could link to that instead.